src/shared_realm.cpp

```diff
@@ -210,7 +210,8 @@ void Realm::open_with_config(const Config& config,
             options.durability = config.in_memory ? SharedGroupOptions::Durability::MemOnly :
                                                     SharedGroupOptions::Durability::Full;
             options.encryption_key = config.encryption_key.data();
-            options.allow_file_format_upgrade = !config.disable_format_upgrade;
+            options.allow_file_format_upgrade = !(config.disable_format_upgrade ||
+                                                  config.schema_mode == SchemaMode::ResetFile);
```

Comment:
IMO `!config.disable_format_upgrade && config.schema_mode != SchemaMode::ResetFile` is slightly more obvious, but this doesn't really matter.